src/woo_publications/utils/tests/test_validators.py

```diff
@@ -0,0 +1,69 @@
+from django.core.exceptions import ValidationError
+from django.test import TestCase, override_settings
+from django.utils.translation import gettext
+
+from woo_publications.publications.tests.factories import TopicFactory
+
+from ..validators import (
+    image_extension_validator,
+    max_file_size_validator,
+    max_file_width_and_height_validator,
+)
+
+
```

Comment:
I don't particularly like these tests but because the validator's need the value to be an `ImageFieldFile` this seems to be the easiest way to do this.

Reply:
they actually don't, they must implement the `ImageFile` interface, which `ImageFieldField` subclasses :)

So, updating your type annotations to use `django.core.files.images.ImageFile` instead should be sufficient, and then You can generate these in memory quite easily:

```py
from io import BytesIO
from django.core.files.images import ImageFile
from PIL import Image

image_file = ImageFile(BytesIO(), name="example.png")
Image.new("RGBA", (16, 16), (255, 0, 0, 255)).save(image_file, format="PNG")
>>> image_file.width
16
>>> image_file.height
16
```